src/python/WMComponent/JobSubmitter/JobSubmitterPoller.py

```diff
@@ -750,23 +747,36 @@ def passSubmitConditions(self):
         """
         _passSubmitConditions_
 
-        Check whether the component is allowed to submit
-        jobs to condor.
+        Check whether the component is allowed to submit jobs to condor.
 
-        Initially it has only one condition, which is the total
-        amount of jobs we can have in condor (pending + running) per schedd,
-        set by MAX_JOBS_PER_OWNER.
+        Initially it has only one condition, which is the total number of
+        jobs we can have in condor (pending + running) per schedd, set by
+        MAX_JOBS_PER_OWNER.
         """
-        passCond = True
-        # fetch idle + running jobs in condor
-        executingJobs = self.countWMBSJobsByState.execute("executing")
-        maxScheddJobs = getScheddParamValue("MAX_JOBS_PER_OWNER")
+        if not self.reqAuxDB:
+            return False
+
+        agentAuxConfig = self.reqAuxDB.getWMAgentConfig(self.hostName)
+        self.condorFraction = agentAuxConfig.get('CondorJobsFraction')
+        if self.condorFraction is None:
+            logging.warning("Failed to retrieve 'CondorJobsFraction' from auxiliar DB")
+            return False
+        else:
+            self.condorFraction = float(self.condorFraction)
+
+        maxScheddJobs = int(getScheddParamValue("MAX_JOBS_PER_OWNER"))
         if maxScheddJobs is None:
-            passCond = False
-        elif int(executingJobs) + self.maxJobsPerPoll >= int(maxScheddJobs):
-            passCond = False
+            logging.warning("Failed to retrieve 'MAX_JOBS_PER_OWNER' from HTCondor")
+            return False
+
+        # fetch idle + running jobs in condor
+        executingJobs = int(self.countWMBSJobsByState.execute("executing"))
+        if executingJobs + self.maxJobsPerPoll >= int(maxScheddJobs * self.condorFraction):
```

Comment:
@ticoann if we have less jobs than `self.maxJobsPerPoll`, then it skips the submission. For instance, if `MAX_JOBS_PER_OWNER` is set to 500 and `self.maxJobsPerPoll` is set to 1000, the component will never submit any jobs.

I think I'll change the code such that if there is 1 single slot available, then we try submitting that job.
What do you think? The case above isn't that bad because these condor limits are usually high, I'm just concerned that it can bug us in the very long future...